src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1441,23 +1431,33 @@ private MemberDeclarationSyntax ParseTypeDeclaration(SyntaxList<AttributeListSyn
 
 #nullable enable
 
-        private TypeDeclarationSyntax ParseClassOrStructOrInterfaceOrRoleDeclaration(SyntaxList<AttributeListSyntax> attributes, SyntaxListBuilder modifiers)
+        private TypeDeclarationSyntax ParseClassOrStructOrInterfaceOrExtensionDeclaration(SyntaxList<AttributeListSyntax> attributes, SyntaxListBuilder modifiers)
```

Comment:
this name seems useless (esp. since it also handles records :)).  How about just ParseTypeDeclaration?

Reply:
records are classes or structs, so arguably they fit in the list.
Renaming to ParseTypeDeclaration results in an error (identical overloads).
ParseTypeDeclarationButEnumOrDelegate? ;-)